File: app/src/main/java/com/github/sewerina/meter_readings/ui/backup_copying/BackupCopyingActivity.java
package com.github.sewerina.meter_readings.ui.backup_copying;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import com.github.sewerina.meter_readings.R;
import com.github.sewerina.meter_readings.ReadingApp;
import com.google.android.material.button.MaterialButton;

public class BackupCopyingActivity extends AppCompatActivity {
    private static final String TAG = "BackupCopyingActivity";
    private BackupCopyingViewModel mViewModel;
    private MaterialButton mBackupMatBtn;
    private MaterialButton mRestoreDbMatBtn;
    private SwipeRefreshLayout mSwipeRefreshLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_backup_copying);

        mSwipeRefreshLayout = findViewById(R.id.swipeRefreshLayout);
        mBackupMatBtn = findViewById(R.id.matBtn_backup);
        mRestoreDbMatBtn = findViewById(R.id.matBtn_restore);

        mViewModel = ViewModelProviders.of(this).get(BackupCopyingViewModel.class);
        ReadingApp.sMainComponent.inject(mViewModel);

        mViewModel.getIsRefreshing().observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean value) {
                mSwipeRefreshLayout.setRefreshing(value);
            }
        });

        mViewModel.getIsAvailable().observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean value) {
                mBackupMatBtn.setEnabled(value);
                mRestoreDbMatBtn.setEnabled(value);
            }
        });

        mViewModel.getMessage().observe(this, new Observer<String>() {
            @Override
            public void onChanged(String message) {
                if (!message.isEmpty()) {
                    Toast.makeText(BackupCopyingActivity.this, message, Toast.LENGTH_SHORT).show();
                }
            }
        });

        mSwipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                mSwipeRefreshLayout.setRefreshing(false);
            }
        });

        mBackupMatBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mViewModel.backup();
            }
        });

        mRestoreDbMatBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mViewModel.restoreDb();
            }
        });


    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

}

File: app/src/main/java/com/github/sewerina/meter_readings/ui/backup_copying/BackupCopyingActivity.kt
package com.github.sewerina.meter_readings.ui.backup_copying

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout.OnRefreshListener
import com.github.sewerina.meter_readings.R
import com.github.sewerina.meter_readings.ReadingApp
import com.google.android.material.button.MaterialButton
import javax.inject.Inject

class BackupCopyingActivity : AppCompatActivity() {
    @Inject
    lateinit var mViewModel: BackupCopyingViewModel

    private lateinit var mBackupMatBtn: MaterialButton
    private lateinit var mRestoreDbMatBtn: MaterialButton
    private lateinit var mSwipeRefreshLayout: SwipeRefreshLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_backup_copying)
        mSwipeRefreshLayout = findViewById(R.id.swipeRefreshLayout)
        mBackupMatBtn = findViewById(R.id.matBtn_backup)
        mRestoreDbMatBtn = findViewById(R.id.matBtn_restore)

        ReadingApp.sMainComponent?.inject(this)
        mViewModel.isRefreshing.observe(this) { value ->
            mSwipeRefreshLayout.isRefreshing = value!!
        }
        mViewModel.isAvailable.observe(this) { value ->
            mBackupMatBtn.isEnabled = value!!
            mRestoreDbMatBtn.isEnabled = value
        }

        mSwipeRefreshLayout.setOnRefreshListener(OnRefreshListener {
            mSwipeRefreshLayout.isRefreshing = false
        })
        mBackupMatBtn.setOnClickListener(View.OnClickListener { mViewModel.backup() })
        mRestoreDbMatBtn.setOnClickListener(View.OnClickListener { mViewModel.restoreDb() })
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    companion object {
        private const val TAG = "BackupCopyingActivity"
    }
}